packages/nimble-components/src/label-provider/base/index.ts

```diff
@@ -0,0 +1,63 @@
+import { DesignToken, FoundationElement } from '@microsoft/fast-foundation';
+import { themeProviderTag } from '../../theme-provider';
+
+/**
+ * Base class for label providers
+ */
+export abstract class LabelProviderBase extends FoundationElement {
+    private readonly deferredTokenUpdates: Map<
+    DesignToken<string>,
+    string | undefined
+    > = new Map();
```

Comment:
I think your suggestions make sense. I had tried something similar in earlier revisions but struggled some with the typing / how to get the property values semi-safely. Got it working so you can take a look now (might need new label-provider-base level tests but haven't done that yet).

A few things I wasn't sure about:
- Is `Observable.getNotifier`+`subscribe` safe to do in the constructor (vs `connectedCallback`)? It seems to work, and it looks like the FAST observable code doesn't necessarily rely on `$fastController`. I think we do need to subscribe before `connectedCallback` (again, for Blazor, the property sets will be before the DOM attach, from what I saw when testing my prototype).
- How to handle `propertyValue === undefined` in handleChange - I ended up deciding the value should be computed from `token.getValueFor(this)` in that case to match what would happen under the initialization codepath.

Also worth noting that the current type for `supportedLabels` ensures the keys are keys/properties on the class, but you'd be allowed to have a key of `$fastController` or something else we inherit from FAST. But the leaf-level tests for the tokens in `label-provider-core.spec.ts` and `label-provider-table.spec.ts` would catch that.

Reply:
Previous questions are N/A with the latest updates. Since we're only tracking the attribute state now I also switched to the more standard pattern of Observable.getNotifier+subscribe in the connectedCallback, and unsubscribe in the disconnectedCallback.